Tools: Add script for checking the development environment requirements

oskosk opened the pull request:
This PR provides a script `tools/check-development-environment.sh` for checking if everything that's needed for working with Jetpack is available on the system.

![image](https://user-images.githubusercontent.com/746152/56892518-48a4c380-6a56-11e9-9f6c-1e6fc7f40439.png)

When you run the script it will check a few things, and if one of the check files, it will provide al ink to a support document where we can help with further details on how to overcome the failed check.

![image](https://user-images.githubusercontent.com/746152/56892682-c668cf00-6a56-11e9-8cf2-654b0b59f81a.png)

![image](https://user-images.githubusercontent.com/746152/56892690-cd8fdd00-6a56-11e9-9221-48e734cd89fc.png)


Fixes nothing

#### Changes proposed in this Pull Request:
<!--- Explain what functional changes your PR includes -->

* Adds script `tools/check-development-environment.sh`.
* Adds document `development-environment-cli-support.md`. which is where we linnk from the CLI to aid in setting up the environment
* ~Adds npm script `yarn check-development-environment`~. Removed it.
* Adds a reference to this script to the document `docs/development-environment.md`.

#### Testing instructions:

1. check out this branch locally
2. Run `tools/check-development-environment.sh.
3. Hopefully all check are green, but if you see one FAILED, confirm the link takes you to an informational section on the CLI Support document online.

#### Proposed changelog entry for your changes:
<!-- Please do not leave this empty. If no changelog entry needed, state as such. -->

* None needed

oskosk committed:
Introduce script to check development environemnt.

The script lives in tools/check-development-environment.sh.

oskosk committed:
Add reference to the script in development-environment.md

oskosk commented:
I'm leaving the support document URL hardcoded to this branch until this is ready to be merged, then I will update to point to master

jeherve commented on docs/development-environment-cli-support.md:
```diff
@@ -0,0 +1,92 @@
+
+### Docker Is Running
+
+make sure the Docker Daemon is running.
```
```suggestion
Make sure the Docker Daemon is running.
```

jeherve commented on docs/development-environment-cli-support.md:
```diff
@@ -0,0 +1,92 @@
+
+If you are using the Jetpack Docker Image, make sure you run `yarn docker:up`.
+
+### Docker Is Running
```
```suggestion
### Docker is running
```

jeherve commented on docs/development-environment-cli-support.md:
```diff
@@ -0,0 +1,92 @@
+
+### Node version is proper
+
+For developing Jetpack you need Node > 10 nowadays.
```
Maybe we shouldn't mention the version number here, since it's most likely to change over time.

jeherve commented on docs/development-environment-cli-support.md:
```diff
@@ -0,0 +1,92 @@
+
+### NVM is available
+
+The `nvm` command is not _really_ needed but it's a tool that allows you to install multiple nvm versions.
```
```suggestion
The `nvm` command is not _really_ needed but it's a tool that allows you to install multiple node versions.
```

jeherve commented on docs/development-environment-cli-support.md:
```diff
@@ -0,0 +1,92 @@
+
+### Command is available: php
+
+PHP is needed all around the build process for Jetpack bundles.
```
Does this take Docker into account?

oskosk replied:
nope

jeherve reviewed (dismissed):
I really like this addition, it could be very helpful for all new contributors! 

I think we could expand a bit on each section (e.g. a link to install Git, nvm, ...). What do you think?

oskosk committed:
Copy updated in doc

oskosk committed:
Update base url to point to master

oskosk commented:
@jeherve I addressed your change requests in 8d26904a814d0cebec9254ce8314077f240304cf.

I plan to continue updating this document after its merged. Need to add images and stuff, but once the base URL for the support is settled in `master`. (Now updated in b131deb25e81c9cf6a24f5103d375ae1f59cab40).

jeherve reviewed (approved):
Let's ship that v1 and ask folks to try it, so we can iterate!

simison commented on tools/check-development-environment.sh:
```diff
@@ -0,0 +1,139 @@
+	assert command_is_available node
+	assert node_version_is_proper
+	assert command_is_available yarn
+	check  nvm_is_available || check command_is_available n
```
This doesn't seem to work — I get:
```
* nvm_is_available : NOPE. Check https://github.com/Automattic/jetpack/blob/master/docs/development-environment-cli-support.md#nvm-is-available
```

I have `n` but not `nvm`:

```
$ n --version
2.1.8

$ nvm --version
-bash: nvm: command not found
```

simison commented on tools/check-development-environment.sh:
```diff
@@ -0,0 +1,139 @@
+
+	assert command_is_available node
+	assert node_version_is_proper
+	assert command_is_available yarn
```
Folks can also have too old Yarn.

Newer versions of Yarn itself check against `engines` values in `package.json` but old ones don't.

You could combine this and `node` versions checks with:

```
npx check-node-version --node --package
npx check-node-version --yarn --package
```

https://www.npmjs.com/package/check-node-version
